pystac_client/stac_api_io.py

```diff
@@ -290,53 +305,19 @@ def get_pages(
                 (link for link in page.get("links", []) if link["rel"] == "next"), None
             )
 
-    def assert_conforms_to(self, conformance_class: ConformanceClasses) -> None:
-        """Raises a :exc:`NotImplementedError` if the API does not publish the given
-        conformance class. This method only checks against the ``"conformsTo"``
-        property from the API landing page and does not make any additional
-        calls to a ``/conformance`` endpoint even if the API provides such an endpoint.
-
-        Args:
-            conformance_class: The ``ConformanceClasses`` key to check conformance
-            against.
-        """
-        if not self.conforms_to(conformance_class):
-            raise NotImplementedError(f"{conformance_class} not supported")
-
-    def conforms_to(self, conformance_class: ConformanceClasses) -> bool:
-        """Whether the API conforms to the given standard. This method only checks
-        against the ``"conformsTo"`` property from the API landing page and does not
-        make any additional calls to a ``/conformance`` endpoint even if the API
-        provides such an endpoint.
-
-        Args:
-            conformance_class : The ``ConformanceClasses`` key to check conformance
-                against.
-
-        Return:
-            bool: Indicates if the API conforms to the given spec or URI.
-        """
-
-        # Conformance of None means ignore all conformance as opposed to an
-        #  empty array which would indicate the API conforms to nothing
-        if self._conformance is None:
-            return True
-
-        class_regex = CONFORMANCE_URIS.get(conformance_class.name, None)
-
-        if class_regex is None:
-            raise Exception(f"Invalid conformance class {conformance_class}")
-
-        pattern = re.compile(class_regex)
-
-        if not any(re.match(pattern, uri) for uri in self._conformance):
-            return False
-
-        return True
-
-    def set_conformance(self, conformance: Optional[List[str]]) -> None:
-        """Sets (or clears) the conformance classes for this StacIO."""
-        self._conformance = conformance
+    @staticmethod
+    def _get_href(
+        obj: pystac.STACObject, rel: str, link: Optional[pystac.Link], endpoint: str
+    ) -> str:
+        if link and isinstance(link.href, str):
+            href = link.absolute_href
+        else:
+            warnings.warn(
+                pystac_client.warnings.MissingLink(rel, obj.__class__.__name__),
+                stacklevel=2,
+            )
+            href = f"{obj.self_href.rstrip('/')}/{endpoint}"
+        return href
```

Comment:
Any reason to attach this function to `StacApiIO`, and not make it a standalone function in a "private" module?

Reply:
no reason at all. I can put it in utils or something

Reply:
>  no reason at all. I can put it in utils or something

Since the first argument is `STACObject`, does it make sense as a mixin that we could include in both `Client` and `CollectionClient` (and anywhere else it is used)?

Reply:
yeah that absolutely makes sense.